DOCS-1220: Add pumpkin board (#1987)
viamrobotics/docs: docs/components/board/customlinux.md

---
title: "Configure a customlinux board"
linkTitle: "customlinux"
weight: 75
type: "docs"
description: "Configure a customlinux board."
images: ["/icons/components/board.svg"]
tags: ["board", "components"]
# SMEs: Susmita
---

Configure a `customlinux` board to integrate linux boards like the [Pumpkin i500](https://ologicinc.com/portfolio/mediateki500/) into your robot.

Complete the following setup requirements, then move on to configuring your board in the [Viam app](https://app.viam.com):

## Setup requirements

- Create a new robot in the [Viam app](https://app.viam.com) and follow the steps on the [setup tab](/installation/#installation-steps) to [install `viam-server`](/installation/) on your `customlinux` board and connect your board to the Viam app.

> - On your `customlinux` board, create a file in your home folder (<file>/home/root</file>) <file>called board.json</file> with your board's pin mappings:

Reviewer: ```suggestion
- On your `customlinux` board, create a file in your (<file>/home/root</file>) directory called <file>board.json</file> with your board's pin mappings:
```
File started in wrong place, otherwise opt

Author: ```suggestion
- On your `customlinux` board, create a file in your <file>/home/root</file> directory called <file>board.json</file> with your board's pin mappings:
```

      "pwm_id": -1
    },
    {
      "name": "16",
      "device_name": "gpiochip0",
      "line_number": 165,
      "pwm_id": -1
    },
    {
      "name": "18",
      "device_name": "gpiochip0",
      "line_number": 1,
      "pwm_id": -1
    },
    {
      "name": "22",
      "device_name": "gpiochip0",
      "line_number": 2,
      "pwm_id": -1
    },
    {
      "name": "24",
      "device_name": "gpiochip0",
      "line_number": 162,
      "pwm_id": -1
    },
    {
      "name": "26",
      "device_name": "gpiochip0",
      "line_number": 0,
      "pwm_id": -1
    },
    {
      "name": "28",
      "device_name": "gpiochip0",
      "line_number": 83,
      "pwm_id": -1
    },
    {
      "name": "32",
      "device_name": "gpiochip0",
      "line_number": 97,
      "pwm_id": -1
    },
    {
      "name": "36",
      "device_name": "gpiochip0",
      "line_number": 151,
      "pwm_id": -1
    },
    {
      "name": "38",
      "device_name": "gpiochip0",
      "line_number": 174,
      "pwm_id": -1
    },
    {
      "name": "40",
      "device_name": "gpiochip0",
      "line_number": 172,
      "pwm_id": -1
    }
  ]
}
```

{{% /tab %}}
{{< /tabs >}}

> The following parameters are available for pin mappings:

Reviewer 2: this feels more customlinux than this particular board, but i dont think it being here is harmful or anything

Reviewer 2: yeah

> The following attributes are available for pin mappings:

Reviewer: ```suggestion
The following fields are available for pin mappings:
```
Would say this or parameters as it's not exactly the same thing as json component attributes right? Not 100% sure

Author: parameters probably even better
```suggestion
The following parameters are available for pin mappings:
```

<!-- prettier-ignore -->
| Name | Type | Inclusion | Description |
| ---- | ---- | --------- | ----------- |
| `name` | string | **Required** | The name of the pin. <br> Example: `"3"`. |

> | `device_name` | string | **Required** | The device name. <br> Example: `"gpiochip0"`. |

Reviewer: If time w/ SME review (understand there might not be), it would be nice to get some explanation on how you find these fields if defining yourself. Why is the pumpkin board's device name `gpiochip0`?

Author: Yeah I am leaving that to the customlinux ticket - to work out because I think there's a lot more detail missing and currently we just want to unblock the pumpkin board

| `line_number` | integer | **Required** | The line number of the pin. <br> Example: `81`. |
| `pwm_id` | integer | **Required** | The power management id of the pin. Example: `-1`. |

## Configure your board

{{< tabs name="Configure a customlinux board" >}}
{{% tab name="Config Builder" %}}

Navigate to the **Config** tab of your robot's page in [the Viam app](https://app.viam.com).
Click on the **Components** subtab and click **Create component**.
Select the `board` type, then select the `customlinux` model.
Enter a name for your `customlinux` board and click **Create**.

![An example configuration for a customlinux board in the Viam app Config Builder.](/components/board/customlinux-ui-config.png)

Copy and paste the following attribute template into your board's **Attributes** box.

> Then fill in the attribute as applicable to your board, according to the table below.

Reviewer: ```suggestion
Then edit the attribute as applicable to your board, according to the table below.
```
(opt) Feel like this makes more sense than fill in in this context
